dev/com.ibm.ws.security.authentication.builtin/src/com/ibm/ws/security/authentication/jaas/modules/LoginModuleHelper.java

```diff
@@ -34,8 +43,7 @@ public static JAASService getJAASService() {
             return testJaasService;
         }
 
-        BundleContext bc = FrameworkUtil.getBundle(IdentityAssertionLoginModule.class).getBundleContext();
-        return bc.getService(bc.getServiceReference(JAASService.class));
+        return jaasService.current().orElseThrow(() -> new NullPointerException("No JAASService found."));
```

Comment:
It is possible that the `jaasService` field may be null here, causing a plain ol' NPE when calling `.current()`.  The field may be null only when ServiceCaller(<init>) throws an ISE due to not obtaining the bundle for class `LoginModuleHelper`. This is extremely unlikely and would otherwise indicate more severe problems. No change requested.

Reply:
Right, I may (later) change the ServiceCaller to not throw an ISA on construction here.  Instead, I would throw ISA when tracking the service the first time the BundleContext is needed from the Bundle.

Reply:
I changed the constructor to no longer throw an ISA.  This makes the unit tests scenario more simple and also allows the code here to be simplified for constructing the ServiceCaller.

Reply:
+1 on improved usability.